### PR TITLE
Buildbot: GCC, refactor, bugs

### PR DESCRIPTION
**Pull request by jfbastien**
Check out GCC, I'll run the torture tests later.
Refactor some of the code.
Fix or comment out some bugs.

### CODE REVIEW

File: buildbot/build.py [hunk 1]
```diff
@@ -51,69 +62,79 @@ def Remove(path):
...
-  print 'Finding prior clang rev'
+  print 'Finding prior Clang rev'
   subprocess.check_call(['git', 'fetch'], cwd=CLANG_DIR)
   prior_rev = FindPriorRev(CLANG_DIR, svn_rev)
-  print 'Checking out clang rev: %s' % prior_rev
-  subprocess.check_call(['git', 'checkout', prior_rev], cwd=CLANG_DIR)
+  print 'Checking out Clang rev: %s' % prior_rev
+  # TODO: broken, should this be:
+  #       ['git', 'svn', 'log', '--show-commit', '--revision=%s' % prior_rev]
+  #subprocess.check_call(['git', 'checkout', prior_rev], cwd=CLANG_DIR)
```
Comment (jfbastien) [inline buildbot/build.py]: @flagxor this seems broken.

File: buildbot/build.py [hunk 2]
```diff
@@ -51,69 +62,79 @@ def Remove(path):
...
-      [os.path.join(CHROMIUM_CLANG_TOOLS_CLANG, 'scripts', 'update.py')])
-  return svn_rev
+        ['git', 'clone', CLANG_GIT, PREBUILT_CLANG_TOOLS_CLANG])
+    GitConfigRebaseMaster(PREBUILT_CLANG_TOOLS_CLANG)
+  print 'Syncing Prebuilt Chromium Clang scripts'
+  subprocess.check_call(['git', 'fetch'], cwd=PREBUILT_CLANG_TOOLS_CLANG)
+  print 'Syncing Prebuilt Chromium Clang'
+  # TODO: this doesn't seem to exist?
+  #subprocess.check_call(
+  #    [os.path.join(PREBUILT_CLANG_TOOLS_CLANG, 'scripts', 'update.py')])
```
Comment (jfbastien) [inline buildbot/build.py]: Ditto broken @flagxor.

File: buildbot/build.py [hunk 3]
```diff
@@ -51,69 +62,79 @@ def Remove(path):
...
   revs = subprocess.check_output(
       ['git', 'rev-list', 'origin/master'], cwd=path).splitlines()
   for rev in revs:
-    num = int(subprocess.check_output(
-        ['git', 'svn', 'find-rev', rev], cwd=path).strip())
-    if num <= goal:
+    num = subprocess.check_output(
+        ['git', 'svn', 'find-rev', rev], cwd=path).strip()
+    if int(num) <= goal:
       return num
```
Comment (jfbastien) [inline buildbot/build.py]: The broken command which uses the return value expects a str, not an int.

File: buildbot/build.py [hunk 4]
```diff
@@ -10,22 +10,33 @@
...
+GCC_GIT = GIT_MIRROR_BASE + 'chromiumos/third_party/gcc'
+SEXPR_GIT = (GIT_MIRROR_BASE + 'external/' +
+             'github.com/WebAssembly/sexpr-wasm-prototype.git')
 
 # Try to use the llvm revision provided by buildbot.
 LLVM_REVISION = os.environ.get('BUILDBOT_REVISION', 'None')
 if LLVM_REVISION == 'None':
   LLVM_REVISION = 'origin/master'
 
+GCC_REVISION = os.environ.get('BUILDBOT_GCC_REVISION', 'None')
```
Comment (jfbastien) [inline buildbot/build.py]: I'm not sure how to get/set this on the bots.

File: buildbot/build.py (hunk 2 above)
Comment (jfbastien) [inline buildbot/build.py]: Fixed!

File: buildbot/build.py (hunk 1 above)
Comment (jfbastien) [inline buildbot/build.py]: Fixed!

File: buildbot/build.py (hunk 3 above)
Comment (jfbastien) [inline buildbot/build.py]: Fixed in d55de7ebc416bbc7645519d1ecbda9e2a24a39cf.

**Comment by flagxor** [comment]
lgtm

**Pull request merged**